### PR TITLE
Clin-1588 Handling multiple regex matches for Jira key

### PR DESCRIPTION
**Pull request by g2-agarwal**
In this PR, we have added a nock package to mock the Jira API calls. This PR also helps to handle multiple Jira keys in a branch name.
We have different scenarios tested for this 
- When all the keys present in the branch are invalid then it will error out like [this](https://github.com/fireflyhealth/lucian/actions/runs/3173077372/jobs/5168284007) 
- When one key is valid in the branch name, it will be successful. [link](https://github.com/fireflyhealth/lucian/actions/runs/3173079012/jobs/5168287475)

### CODE REVIEW

File: __tests__/jira.test.ts [hunk 1]
```diff
@@ -110,13 +111,72 @@ describe('getLabelsForDisplay()', () => {
...
+      });
+
+    // for mocking the request for 404 request
+    nock('https://cleartaxtech.atlassian.net/rest/api/3')
+      .get('/issue/TEST-123456?fields=project,summary,issuetype,labels,status,customfield_10016')
+      .reply(404, {});
+    const jira = new Jira('https://cleartaxtech.atlassian.net', '<username>', '<token_here>');
+    // when the first key is valid
+    const details = await jira.getJiraDetails(['TEST-1234', 'TEST-123456']);
     expect(details).not.toBeNull();
```
Comment (shivananda-sahu) [inline __tests__/jira.test.ts:142]: should we assert the contents of `details`

File: __tests__/jira.test.ts [hunk 2]
```diff
@@ -110,13 +111,72 @@ describe('getLabelsForDisplay()', () => {
...
+
+    // for mocking the request for 404 request
+    nock('https://cleartaxtech.atlassian.net/rest/api/3')
+      .get('/issue/TEST-123456?fields=project,summary,issuetype,labels,status,customfield_10016')
+      .reply(404, {});
+    const jira = new Jira('https://cleartaxtech.atlassian.net', '<username>', '<token_here>');
+
+    // when the second key is a valid
+    const details = await jira.getJiraDetails(['TEST-123456', 'TEST-1234']);
+    expect(details).not.toBeNull();
```
Comment (shivananda-sahu) [inline __tests__/jira.test.ts:175]: should we assert the contents of `details`

File: __tests__/jira.test.ts [hunk 3]
```diff
@@ -110,13 +111,72 @@ describe('getLabelsForDisplay()', () => {
...
+
+    // when both the keys are wrong
+    const result = await jira.getJiraDetails(['TEST-123456', 'TEST-12341']);
+    expect(result).toBeNull();
```
Comment (shivananda-sahu) [inline __tests__/jira.test.ts:180]: should we assert the contents of `details`

File: __tests__/jira.test.ts (hunk 1 above)
Comment (g2-agarwal) [reply __tests__/jira.test.ts:142]: done

File: __tests__/jira.test.ts (hunk 2 above)
Comment (g2-agarwal) [reply __tests__/jira.test.ts:175]: done

File: __tests__/jira.test.ts (hunk 3 above)
Comment (g2-agarwal) [reply __tests__/jira.test.ts:180]: No need to assert the content as the "result" will be null because both the keys are wrong.

**Review by shivananda-sahu — APPROVED** [review]

**Pull request merged**